staging/src/k8s.io/apimachinery/pkg/util/validation/field/path.go

```diff
@@ -69,6 +69,17 @@ func (p *Path) Root() *Path {
 	return p
 }
 
+// SetRoot returns a Path with root replaced.
```

Comment:
the implementation doesn't really seem to match the docs... it's not actually replacing the root. I also don't think we really need this, see comment around construction/use of NormalizedRuleFieldPath